### PR TITLE
added research hub pages to percy

### PR DESCRIPTION
**Pull request by danielfmiranda**
## Description

Link to sample test page: (Please refer to percy tests in the CI panel)
Related PRs/issues: #10629

This PR adds the following pages to be tested by percy:
- Research Hub Landing page
- Research Hub Library Page
- Research Hub Detail page
- Research Hub Author Index Page
- Research Hub Author Detail Page

As well as some minor changes to the factory that were needed in order to accomodate the tests above.

## Steps to test

1. Observe the Percy tests for this PR, note that all page types listed above are now being tested.
2. If everything looks as expected, testing is complete!

### CODE REVIEW

**Comment by mmmavis** [comment]
Hi @danielfmiranda , Percy screenshots show visual differences unrelated to RCC pages. Do you know why? I'm curious if it's something caused by seeding.

File: network-api/networkapi/wagtailpages/factory/libraries/research_hub/__init__.py [hunk 1]
```diff
@@ -18,6 +18,20 @@
 from networkapi.wagtailpages.factory.libraries.research_hub import (
     taxonomies as taxonomies_factory,
 )
+from networkapi.wagtailpages.pagemodels.profiles import Profile
+
+
+def create_detail_page_for_visual_regression_tests(seed, research_library_page):
+    percy_author_profile = Profile.objects.get(name="Percy Profile")
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/factory/libraries/research_hub/__init__.py:25]: `seed` didn't get used here. Did you intend to use it to reseed?

Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/factory/libraries/research_hub/__init__.py:27]: Thanks for the catch Mavis! I have added `reseed(seed)` to the file on the latest commit of the branch.

**Review by mmmavis — APPROVED** [review]
Hey @danielfmiranda , after some detective work I'm guessing the visual changes are due to the new profile addition (Percy Profile) we are adding in this PR. If that's the case then these visual changes are expected because the dataset got changed hence different set of objects are selected. I think we can go ahead and merge this PR. If the new factory code is causing more visual changes on all other PRs, we can still revert the merge and investigate further to find a solution. How does this sound to you?

**Comment by danielfmiranda** [comment]
Hi @mmmavis, thanks for taking a deeper look! 

I gave it another look as well, and can confirm that a majority of these changes are in fact just small changes to either randomly generated text or author profile relationships.

However, I did find one area that had room for improvement and that was on the "Privacy Not Included category page", where there was quite a big difference in the number of "related articles". In the screenshot below, we can see that before this PR the page had 6, and after the PR the page has 1.

This was due to the fact that the factory for this "related article" relationship was using the param `max_count`, which in this case would generate a random number of related articles, with a maximum of 6. 


Updating this to use the parameter of `exact_count`, ensures that the percy page always has related articles. And while they are still can be randomly generated, the visual diffs should be much smaller than before :+1:

# Before:

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/1154136e5d7b0cb9c87d7d429d5b4a98439c787d/network-api/networkapi/wagtailpages/factory/buyersguide.py#L514-L521

Screenshot 2023-05-25 at 18-09-57 Build #14671 - Mozilla Foundation - Percy Visual testing as a service


# After:

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/f23c5500d6341937d96ed804cb78ff0e006f371d/network-api/networkapi/wagtailpages/factory/buyersguide.py#L514-L521



Screenshot 2023-05-25 at 18-08-40 Build #14673 - Mozilla Foundation - Percy Visual testing as a service

**Review by mmmavis — APPROVED** [review]
LGTM!

**Pull request merged**